calico_containers/adapter/datastore.py

```diff
@@ -794,12 +914,28 @@ def get_endpoints(self, hostname, container_id):
         # Extract all of the endpoints.
         eps = []
         for endpoint in endpoints:
-            (_, _, _, _, _, _, _, _, _, endpoint_id) = \
-                                                 endpoint.key.split("/", 9)
-            eps.append(Endpoint.from_json(endpoint_id, endpoint.value))
+            eps.append(Endpoint.from_json(endpoint.key, endpoint.value))
         return eps
 
     @handle_errors
+    def get_endpoint_from_id(self, endpoint_id):
```

Comment:
This should really just use query_endpoint.

Raise an Exception if >1 endpoint matching the endpoint ID.